tests/e2e-pw/playwright.config.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+/**
+ * External dependencies
+ */
+import { defineConfig, devices } from '@playwright/test';
+
+/**
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// require('dotenv').config();
+
+const { BASE_URL } = process.env;
+
+/**
+ * See https://playwright.dev/docs/test-configuration.
+ */
+export default defineConfig( {
+	testDir: './specs/',
+	/* Run tests in files in parallel */
+	fullyParallel: true,
+	/* Fail the build on CI if you accidentally left test.only in the source code. */
+	forbidOnly: !! process.env.CI,
+	/* Retry on CI only */
+	retries: process.env.CI ? 2 : 0,
+	/* Opt out of parallel tests on CI. */
+	workers: process.env.CI ? 1 : undefined,
+	/* Reporter to use. See https://playwright.dev/docs/test-reporters */
+	reporter: 'html',
+	outputDir: './test-results',
+	/* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+	use: {
+		baseURL: BASE_URL ?? 'http://localhost:8084',
+		screenshot: 'only-on-failure',
+		trace: 'retain-on-failure',
+		video: 'on-first-retry',
+		viewport: { width: 1280, height: 720 },
+		// storageState: 'tests/e2e-pw/storage/state.json',
+	},
+	expect: {
+		toHaveScreenshot: { maxDiffPixels: 100 },
+	},
+	snapshotPathTemplate:
+		'{testDir}/__snapshots__/{testFilePath}/{arg}_{projectName}_{platform}{ext}',
```

Comment:
`{platform}` results in snapshots that aren't compatible across darwin/linux.

E.g. on macOS, the filename is `Merchant-deposits-Load-the-deposits-list-page-1_chromium_darwin.png`
When the test runs on GH actions, it expects a file `Merchant-deposits-Load-the-deposits-list-page-1_chromium_linux.png`

Should we remove the `{platform}` part of snapshot filenames? This would require screenshots being close enough across both darwin/linux.

Reply:
I'll test this in 6897f72180a6babfe9ecf584d386dd6e3a9ba239 and see if the GH checks on #7922 pass

Reply:
Could also recommend/require running tests in known env (e.g. docker container running linux).

Reply:
It might work, but the GH check e2e env has deposit history that the local env doesn't have:

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/8e537c41-c138-4b17-8deb-764099fb8202)

Reply:
Aha, so ideally that history would be standardised in the tests container (sounds like this snapshot depends on leaky state and could be more self-contained).

Reply:
>Should we remove the {platform} part of snapshot filenames? This would require screenshots being close enough across both darwin/linux.

Yep, I think we can get rid of the platform part aiming for a common look. We can always adjust the pixel diff to fit slight differences by fonts or similar things.

>It might work, but the GH check e2e env has deposit history that the local env doesn't have

🤔 That makes it trickier, ideally, we should try to have the same env. In the past I used a fully mocked server for visual regression testing, to remove it from the equation, but that will require another big effort. We can try to limit screenshots to more generic parts, for now.

Reply:
> We can try to limit screenshots to more generic parts, for now.

Yeh, agree we can keep it simple for the MVP of the Playwright PR.

Reply:
I've added a non-state-dependant UI test in a25a555 – mainly to see how different this will be cross-platform, but also this UI was an element that broke in the past due to a wp/components update IIRC.

Reply:
> see how different this will be cross-platform

The [snapshot test failed](https://github.com/Automattic/woocommerce-payments/actions/runs/7498234661?pr=7671) due to small font-rendering differences. 

```
3533 pixels (ratio 0.01 of all image pixels) are different.
```

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/e7c581e0-d264-4262-89d2-e1a258b27efc)

Linux:
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/e8f62434-0415-4f99-8179-3c2805d49bca)

Darwin:
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/827ce09d-bf0a-421e-a879-f6e65ba06c79)

Reply:
I've adjusted the `maxPixelDiffRatio` to `0.025` in 889b5b5 (> `0.01` which failed the cross-platform test). See how this goes...

Reply:
Yes, I was expecting fonts to be an issue between platforms. I usually solve that by using a specific font instead of the system one. Do you know if we could use a linux container and still use the Playwright UI? 🤔

Reply:
[Running in UI mode from within a Docker container is mentioned in the docs](https://playwright.dev/docs/test-ui-mode#docker--github-codespaces), so might be worth a shot if we think it'd be more reliable – hopefully it doesn't add too much complexity.

There is a [docs page with the recommended path for running browsers in a Docker container](https://playwright.dev/docs/docker).

Reply:
Getting the network right is proving painful with Docker:

Running this command is almost working, and the PW UI is available at `localhost:8077`/

But then PW only gets a buggy version of WP and cannot login to wp-admin. I think we need to tell WP that the site's hostname is `host.docker.internal:8084`.

```
docker run -v -p 8077:8077 $PWD:/woopayments -w /woopayments -e BASE_URL=http://host.docker.internal:8084 -it --rm --ipc=host mcr.microsoft.com/playwright:v1.39.0-jammy npm run test:e2e-pw -- --ui-host=0.0.0.0 --ui-port=8077
```

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/b543f96d-b428-4d97-a481-eb45d7c33b0d)

Reply:
As mentioned in Slack, I think we leave the Docker env part out of this PR's scope and handle it as a follow-up issue. Everything works without running in Docker, although it may make the visual regression tests more reliable.

p1704929040720799/1702046252.009489-slack-C053VQPD74J

Reply:
Revisiting visual regression testing, I added one for this dispute notice component – font rendering differences will make visual testing unreliable until we fix this.

Should we fix this for this MVP PR or leave it as a followup issue?

<img width="966" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/14054f3a-f127-4e28-8c4e-3595f59104bf">

Reply:
🏗️ I've got to a point where running in a Docker env locally works, and the visual snapshots match with the GH Actions snapshot.

To do this:

1. I added the following to `tests/e2e/docker/wordpress/wp-config.php`:

		```php
		define('DOCKER_REQUEST_URL', (!empty($_SERVER['HTTPS']) ? 'https://' : 'http://') . (!empty($_SERVER['HTTP_HOST']) ? $_SERVER['HTTP_HOST'] : 'localhost'));
		define('WP_SITEURL', DOCKER_REQUEST_URL);
		define('WP_HOME', DOCKER_REQUEST_URL);
		```

2. Then Run the following commands:
	- UI mode: `docker run -v $PWD:/woopayments -w /woopayments -p 8077:8077 -e BASE_URL=http://host.docker.internal:8084 -it --rm --ipc=host mcr.microsoft.com/playwright:v1.39.0-jammy npm run test:e2e-pw -- --ui-host=0.0.0.0 --ui-port=8077`
	- CLI run: `docker run -v $PWD:/woopayments -w /woopayments -e BASE_URL=http://host.docker.internal:8084 -it --rm --ipc=host mcr.microsoft.com/playwright:v1.39.0-jammy npm run test:e2e-pw`


So if we tidy this up into the npm setup & run scripts, we'll have a better setup for robust visual regression tests.

Reply:
This thread is getting long, and the Docker part of this may not make it into this MVP PR.

So I've opened #8160 to implement running PW locally from a linux Docker container.

Reply:
Resolved now that #8160 has been merged into this branch.